File: Algorithm/Programmers/Level2/solution(8).py
# 위장

def solution(clothes):
    answer = 1
    dic = {}

    
    for i in range(len(clothes)):
        if clothes[i][1] in dic:
            dic[clothes[i][1]].append(clothes[i][0])
        else:
            dic[clothes[i][1]] = [clothes[i][0]]
    
    for value in dic.values():
        answer *= len(value) + 1 # 해당 종류의 의상을 입지 않는 경우 포함
    
    return answer - 1 # 안 입은 경우 제외

File: Algorithm/Programmers/Level1/solution(8).java
// 나누어 떨어지는 숫자 배열

import java.util.*; 

class Solution {
    public int[] solution(int[] arr, int divisor) {
        int[] answer = {};
        ArrayList<Integer> arrlist = new ArrayList<Integer>();
        for (int i = 0; i < arr.length; i++){
            if (arr[i] % divisor == 0)
                arrlist.add(arr[i]);
        }
        
        if (arrlist.size() == 0)
            arrlist.add(-1);
        
        int n = arrlist.size();
        answer = new int[n];
        
        for (int i = 0; i < n; i++){
            answer[i] = arrlist.get(i);
        }
        
        Arrays.sort(answer);
        
        return answer;
    }
}